packages/core/js-client/src/particle/Particle.ts

```diff
@@ -47,12 +52,28 @@ export class Particle implements IParticle {
             toUint8Array(json.data),
             json.ttl,
             json.init_peer_id,
+            toUint8Array(json.signature)
         );
 
         return res;
     }
 }
 
+/**
+ * Builds particle message for signing
+ */
+export const buildParticleMessage = ({ id, initPeerId, timestamp, ttl, script }: Omit<IParticle, 'signature' | 'data'>): Uint8Array => {
+    const en = new TextEncoder();
+
+    return concat([
+        en.encode(id),
+        en.encode(initPeerId),
```

Comment:
You should remove init_peer_id, look at the pr [description](https://github.com/fluencelabs/nox/pull/1811). There is no need in `init_peer_id`, because signature contains it implicitly

Reply:
https://www.notion.so/fluencenetwork/Particle-Signature-ebd119a06b5248448465bbb07d759f1e
I was implementing signatures following this doc

Reply:
I have written this doc. So lets change it, because it has really no sense to put init_peer_id in signature